packages/core/nuxt-theme-module/theme/components/WishlistSidebar.vue

```diff
@@ -100,25 +120,50 @@ export default {
     SfCollectedProduct,
     SfImage
   },
-  setup() {
+  setup(props, { root }) {
     const { isWishlistSidebarOpen, toggleWishlistSidebar } = useUiState();
-    const { wishlist, removeItem, load: loadWishlist } = useWishlist();
+    const { wishlist, removeItem: removeFromWishlist, load: loadWishlist, clear: clearWishlist } = useWishlist();
     const { isAuthenticated } = useUser();
+    const { addItem } = useCart();
     const products = computed(() => wishlistGetters.getItems(wishlist.value));
     const totals = computed(() => wishlistGetters.getTotals(wishlist.value));
     const totalItems = computed(() => wishlistGetters.getTotalItems(wishlist.value));
 
+    watch(() => root.$route, () => {
+      if (isWishlistSidebarOpen.value) toggleWishlistSidebar();
+    });
+
+    const addToCart = async (product) => {
+      await addItem({
+        product: { sku: product.variant.sku },
+        quantity: 1
+      });
+      removeFromWishlist({ product });
+    };
+
+    const addAllToCart = async () => {
+      await Promise.all(
```

Comment:
This is not the best approach for adding bulk products to cart because if we have like a 10 products in our wishlist and then press this button we would fire 10 async requests to the backend. Could you please check if we have a more efficient way of adding to cart (something like a bulk add to cart)?

Reply:
@Baroshem I don't think we have too many existing options here. However, I could easily create such an api method from scratch which would accept multiple products and create multiple `addLineItem` actions from them. That would allow us to add the new functionality in a neat, clean and performance-saving way, without any breaking changes.
```ts
interface ProductPayload {
  product: ProductVariant,
  quantity: Number
}

addMultipleToCart(
  settings,
  { id, version }: CartDetails,
  products: ProductPayload[],
  customQuery?: CustomQuery
 ) => Promise<CartResponse>
 ```

Reply:
I see. This would require an RFC I believe because it is a completely new feature. @andrzejewsky what do you think about it?

Reply:
Same for `clearCart` which is mocked at the moment. It would require a similar approach. I could implement both.

Reply:
such logic shouldn't be a part of the front-end... you are implemented to some extent business logic here. That's supposed to be implemented on the backend-side only and as front-end dev, you would need to know that when you call a specific endpoint, it will transfer your product from the current wishlist to cart items. 

Additionally, you are implementing the custom feature, that's not a part of comemrcetools at all, while VSF is the only client of that certain platform. If you want to add something like this, you can create an api-client function that will do than transition.

What's more, please next time do a research of the API you integrate with, because quick check and you can add all of the items within one request, instead of iterating and making separate calls

Reply:
> Additionally, you are implementing the custom feature, that's not a part of comemrcetools at all, while VSF is the only client of that certain platform. If you want to add something like this, you can create an api-client function that will do than transition.

> What's more, please next time do a research of the API you integrate with, because quick check and you can add all of the items within one request, instead of iterating and making separate calls

@andrzejewsky Correct me if I'm wrong, but isn't that what I had proposed here?

> @Baroshem I don't think we have too many existing options here. However, I could easily create such an api method from scratch which would accept multiple products and create multiple addLineItem actions from them. That would allow us to add the new functionality in a neat, clean and performance-saving way, without any breaking changes.

Reply:
@lsliwaradioluz yup, just posted in general what we can proceed with scenarios like this one

Reply:
@andrzejewsky Ok, so basically doing it like I did above (with the Promise.all() thing) is definitely a no-no - agreed. However, is there a green light for implementing those new api-client methods, namely `addMultipleToCart` and `clearCart` ? I just want to make sure.

Reply:
you can create just `addLineItems` and accept array, so you can reuse that in adding item in current composable, and also developers will be able to use that function later in the code if there is a need

Reply:
and ofc i would suggest to create another api-client function that will transfer wishlist items to cart items

Reply:
Guys take a look at it, it's changed now.